compiler+runtime/include/cpp/jank/runtime/obj/uuid.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <boost/uuid/uuid.hpp>
+
+#include <jank/runtime/object.hpp>
+
+namespace jank::runtime::obj
+{
+  struct uuid : gc
+  {
+    static constexpr object_type obj_type{ object_type::uuid };
+    static constexpr native_bool pointer_free{ false };
+
+    uuid();
+    uuid(native_persistent_string const &s);
+
+    /* behavior::object_like */
+    native_bool equal(object const &) const;
+    native_persistent_string to_string() const;
+    void to_string(util::string_builder &buff) const;
+    native_persistent_string to_code_string() const;
+    native_hash to_hash() const;
+
+    object base{ obj_type };
+
+    boost::uuids::uuid value{};
```

Comment:
I went with this representation as I assume boost has some internal compact representation, however I did not verify. Also, not sure what to call this field.

Reply:
Internal storage: 128-bit data stored as a byte array  - according to google 😂

Reply:
Actually I can just goto definition 🤦 
```cpp
using repr_type = std::uint8_t[ 16 ];
```

Reply:
This is just fine. If the uuid header shows up as a heft include, we can later change the struct to store something opaque and only include boost's uuid in the cpp file. At this point, I don't think we can do that without affecting runtime perf.